src/renderers/WebGLRenderer.js

```diff
@@ -905,19 +907,13 @@ function WebGLRenderer( parameters ) {
 
 						const material2 = material[ i ];
 
-						if ( compiled.has( material2 ) === false ) {
```

Comment:
I'm afraid this check does not work anymore since materials can now be compiled multiple times.

Reply:
The change seems otherwise acceptable, but this code doesn't seem to update materialProperties.currentProgram, which prevents early outs in getProgram().